processing/src/main/java/org/apache/druid/utils/ConnectionUriUtils.java

```diff
@@ -177,19 +178,26 @@ public static Set<String> tryParsePostgresConnectionUri(String connectionUri)
   }
 
   public static Set<String> tryParseMySqlConnectionUri(String connectionUri)
-      throws ClassNotFoundException, NoSuchMethodException, InstantiationException, IllegalAccessException,
+      throws ClassNotFoundException, NoSuchMethodException, IllegalAccessException,
              InvocationTargetException
   {
-    Class<?> driverClass = Class.forName(MYSQL_NON_REGISTERING_DRIVER);
-    Method parseUrl = driverClass.getMethod("parseURL", String.class, Properties.class);
-    // almost the same as postgres, but is an instance level method
-    Properties properties = (Properties) parseUrl.invoke(
-        driverClass.getConstructor().newInstance(),
-        connectionUri,
-        null
-    );
-
-    if (properties == null) {
+    Class<?> connectionUrlParserClass = Class.forName(MYSQL_CJ_CONF_CONNECTION_URL_PARSER);
+    Method parseUrl = connectionUrlParserClass.getMethod("parseConnectionString", String.class);
+    // almost the same as postgres
+    Object connectionUrlParser = parseUrl.invoke(null, connectionUri);
+    Method getProperties = connectionUrlParserClass.getMethod("getProperties");
+    Properties properties = new Properties();
+    Map<String,String> propertyMap =  (Map<String, String>) getProperties.invoke(connectionUrlParser);
+    for (Map.Entry<String, String> entry : propertyMap.entrySet()) {
+      if (entry.getKey() != null) {
+        if(entry.getValue() == null) {
```

Comment:
is this actually possible?

Reply:
Yes, it is possible. Like in this [case](https://github.com/apache/druid/blob/e373f6269251655f5be93ce895aee8dee8cc67dd/processing/src/test/java/org/apache/druid/utils/ConnectionUriUtilsTest.java#L42) where `keyonly` doesn't have any value associated with it.